nncf/quantization/algorithms/accuracy_control/evaluator.py

```diff
@@ -0,0 +1,176 @@
+            # - torch.tensor([1.0], dtype=torch.float32)
+            # - tf.constant(1.0, dtype=tf.float32
+            # - tf.constant([1.0], dtype=tf.float32)
+            values_for_each_item = [float(x) for x in values_for_each_item]
+
+        return float(metric), values_for_each_item
+
+    def _determine_mode(self, model_for_inference: TModel, dataset: Dataset) -> None:
+        """
+        Determines mode based on the type of returned value from the
+        validation function.
+
+        :param model_for_inference: Model to validate.
+        :param dataset: Dataset to validate the model.
+        """
+        data_item = dataset.get_data([0])
+        # pylint: disable=W0703
+        try:
+            metric_value, values_for_each_item = self._validation_fn(model_for_inference, data_item)
+        except Exception:
+            self._metric_mode = False
+
+        if self._metric_mode is not None:
+            return
+
+        try:
+            metric_value = metric_value if metric_value is None else float(metric_value)
+        except Exception as ex:
+            raise RuntimeError(
+                f"Metric value of {type(metric_value)} type was returned from the `validation_fn` "
+                "but the float value is expected."
+            ) from ex
+
+        convert_to_float_possible = True
+        if values_for_each_item is not None:
+            # pylint: disable=W0703
+            try:
+                _ = float(values_for_each_item[0])
+            except Exception:
+                convert_to_float_possible = False
+
+        # Analyze `metric_value` and `values_for_each_item` values:
+        # +--------------+----------------------+-------------+
+        # | metric_value | values_for_each_item | metric_mode |
+        # +--------------+----------------------+-------------+
+        # | float        | None                 | True        |
+        # +--------------+----------------------+-------------+
+        # | float        | List[float]          | True        |
+        # +--------------+----------------------+-------------+
+        # | float        | List[List[TTensor]]  | False       |
+        # +--------------+----------------------+-------------+
+        # | None         | None                 | False       |
+        # +--------------+----------------------+-------------+
+        # | None         | List[float]          | UNEXPECTED  |
+        # +--------------+----------------------+-------------+
+        # | None         | List[List[TTensor]]  | False       |
+        # +--------------+----------------------+-------------+
+
+        self._metric_mode = False
+        if isinstance(metric_value, float) and (values_for_each_item is None or convert_to_float_possible):
+            self._metric_mode = True
+        elif values_for_each_item is not None and not isinstance(values_for_each_item[0], list):
+            raise RuntimeError("Unexpected return value from provided validation function.")
```

Comment:
From the user's perspective, this message is not clear. What should they do in this case?

Reply:
They should refer to the documentation to get more details about what validation function should return.